File: src/dev/pitlor/rider_service_fabric_support/utils/TextFieldChangeListener.java
package dev.pitlor.rider_service_fabric_support.utils;

import javax.swing.event.DocumentEvent;
import javax.swing.event.DocumentListener;

@FunctionalInterface
public interface TextFieldChangeListener extends DocumentListener {
	void onChange(DocumentEvent e);

	@Override
	default void insertUpdate(DocumentEvent e) {
		onChange(e);
	}

	@Override
	default void removeUpdate(DocumentEvent e) {
		onChange(e);
	}

	@Override
	default void changedUpdate(DocumentEvent e) {
		onChange(e);
	}
}

File: src/main/kotlin/dev/pitlor/rider_service_fabric_support/swing_components/helpers/TextFieldChangeListener.kt
package dev.pitlor.rider_service_fabric_support.swing_components.helpers

import javax.swing.event.DocumentEvent
import javax.swing.event.DocumentListener

@FunctionalInterface
class TextFieldChangeListener(private val onChange: (DocumentEvent) -> Unit) : DocumentListener {
    override fun insertUpdate(e: DocumentEvent) {
        onChange(e)
    }

    override fun removeUpdate(e: DocumentEvent) {
        onChange(e)
    }

    override fun changedUpdate(e: DocumentEvent) {
        onChange(e)
    }
}